dkist/conftest.py

```diff
@@ -347,3 +347,14 @@ def visp_dataset_no_headers(tmp_path_factory):
         with open(vispdir / "test_visp_no_headers.asdf", mode="wb") as afo:
             afo.write(gfo.read())
     return load_dataset(vispdir / "test_visp_no_headers.asdf")
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_call(item):
+    dslist = item.config.getoption("--ds")
+    dslist = dslist.split(",") if dslist else None
+    if dslist and item.get_closest_marker("accept_cli_datasets"):
+        dsargs = [load_dataset(ds) for ds in dslist]
+        for ds, arg in zip(dsargs, item.fixturenames):
+            item.funcargs[arg] = ds
```

Comment:
This is letting you pass in multiple datasets which get passed as multiple *different* fixtures to the tests, or passed in as multiple parametrized inputs to a single fixture? I feel like the second option is the most useful?

Reply:
It's passing them in as different fixtures, although I'm not sure that as written it will actually work for eg `some_test(fixture_a, fixture_b`) if `fixture_a` depends on other fixtures, because of how pytest orders them in the list of arguments. Your suggestion of modifying the marker should fix that though.

I think passing in parametrized inputs to a single fixture could be useful but I think that's a different task, unless I've misunderstood you.

Reply:
Do we need to support more than one fixture per test? surely almost all of them only use one?

Reply:
Almost all do, yes, but a few use two, for example to compare more complex datasets to the base one. Those ones probably do want to be reworked a bit with this anyway so it may be that we can refactor out needing more than one fixture input.

Reply:
Do we need to support that use case here? feels too niche?

Reply:
Probably not, I'm happy to just leave it